### DISCUSSION TITLE
rm `PolicyId`

### DISCUSSION THREAD

**Discussion by waalge** [aiken-lang/stdlib#101] [General]
The era of first class multi purpose scripts has made clear that having a distinct type for this concept is a mistake. 
It obscures the natural identification of a token with its credential/ address(es)

**Comment by logicalmechanism** [comment]
Couldn't you just type it as ScriptHash when you want to make a distinction?

I use PolicyId when the context is a value and ScriptHash when the context is a script address/credential. Personally, I find it very use case dependent and typically just use PolicyId / ScriptHash for readability and not natural identification.

**Reply by waalge** [reply]
> Couldn't you just type it as ScriptHash when you want to make a distinction?

You mean when you want to make the correspondence? Yes and IMO this should be the default.
Having more than one type for the same thing is baggage. \

Ultimately, this an aesthetics thing - aiken does not distinguish type aliases so it could just be `ByteArray`.
My point is: multipurpose validators are now _first class_, thus ScriptHash should be the consistent type regardless of the purpose.
It is meaningful to say

```
  expect Mint(script_hash) = purpose 
  expect [output, ..] = outputs
  expect Script(script_hash) == output.address.payment_credential   
```

Should It be "natural" to do an equality like this? I'm saying yes and that this should not be obscured by the types (aliases). 
We do not distinguish the type of credentials in the other script purposes. Withdraw and Certificates all use `Credential`.
